lib/internal/Magento/Framework/Reflection/TypeProcessor.php

```diff
@@ -708,4 +695,58 @@ private function getNormalizedType($type)
         }
         return $type;
     }
+
+    /**
+     * Parses `return` annotation from reflection method.
+     *
+     * @param MethodReflection $methodReflection
+     * @return ReturnTag
+     * @throws \InvalidArgumentException if doc block is empty or `@return` annotation doesn't exist
+     */
+    private function getMethodReturnAnnotation(MethodReflection $methodReflection)
+    {
+        $methodName = $methodReflection->getName();
+        $returnAnnotations = $this->getReturnFromDocBlock($methodReflection);
+        if (empty($returnAnnotations)) {
+            // method can inherit doc block from implemented interface, like for interceptors
+            $implemented = $methodReflection->getDeclaringClass()->getInterfaces();
+            /** @var ClassReflection $parentClassReflection */
+            foreach ($implemented as $parentClassReflection) {
+                if (!$parentClassReflection->hasMethod($methodName)) {
+                    continue;
+                }
+                $returnAnnotations = $this->getReturnFromDocBlock(
+                    $parentClassReflection->getMethod($methodName)
+                );
+
+                break;
+            }
+            // throw an exception if even implemented interface doesn't have return annotations
+            if (empty($returnAnnotations)) {
+                throw new \InvalidArgumentException(
+                    "Getter return type must be specified using @return annotation. "
+                    . "See {$methodReflection->getDeclaringClass()->getName()}::{$methodName}()"
+                );
+            }
+        }
+        return $returnAnnotations;
+    }
+
+    /**
+     * Parses `return` annotation from doc block.
+     *
+     * @param MethodReflection $methodReflection
+     * @return ReturnTag
+     */
+    private function getReturnFromDocBlock(MethodReflection $methodReflection)
+    {
+        $methodDocBlock = $methodReflection->getDocBlock();
+        if (!$methodDocBlock) {
+            throw new \InvalidArgumentException(
+                "Each getter must have dock block. "
```

Comment:
Typo dock -> doc. "have **a** doc block"?

Reply:
Done

Reply:
> +                "Each getter must have a dock block. "

I meant don't see this.

Reply:
https://github.com/magento/magento2/pull/11556/files#diff-292261de777138aa6bb4342b9757a3a5R744

Reply:
"dock" is correct spelling? I was not certain regarding "a" but "dock" seems to be not correct.

Reply:
Yes, sure. Fixed.